File: chazm-model/src/main/java/runtimemodels/chazm/model/event/InstanceGoalEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.InstanceGoal;

/**
 * The {@linkplain InstanceGoalEventFactory} interface defines the API for constructing {@linkplain InstanceGoalEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface InstanceGoalEventFactory {

    /**
     * Constructs an {@linkplain InstanceGoalEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param goal     the {@linkplain InstanceGoal}.
     * @return a {@linkplain InstanceGoalEvent}.
     */
    InstanceGoalEvent build(EventType category, InstanceGoal goal);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/InstanceGoalEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.InstanceGoal

/**
 * The [InstanceGoalEventFactory] interface defines the API for constructing [InstanceGoalEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface InstanceGoalEventFactory {

    /**
     * Constructs an [InstanceGoalEvent].
     *
     * @param category the [EventType].
     * @param goal     the [InstanceGoal].
     * @return a [InstanceGoalEvent].
     */
    fun build(category: EventType, goal: InstanceGoal): InstanceGoalEvent

}
